src/main/common/time.c

```diff
@@ -120,14 +122,80 @@ static bool rtcIsDateTimeValid(dateTime_t *dateTime)
            (dateTime->millis <= 999);
 }
 
-static void dateTimeWithOffset(dateTime_t *dateTimeOffset, dateTime_t *dateTimeInitial, int16_t minutes)
+static int lastSundayOfMonth(int currentYear, int wantedMonth) {
+    int days[] = {31,29,31,30,31,30,31,31,30,31,30,31};
+	int m, y = currentYear, w;
+    days[1] -= (y % 4) || (!(y % 100) && (y % 400));
+    w = y * 365 + (y - 1) / 4 - (y - 1) / 100 + (y - 1) / 400 + 6;
+
+	for(m = 0; m < 12; m++) {
+		w = (w + days[m]) % 7;
+        if (m == wantedMonth - 1) {
+            return days[m] + (w < 5 ? -2 : 5) - w;
+        }
+	}
+    return 0;
+}
+
+static int nthSundayOfMonth(int lastSunday, int nth) {
+    while (lastSunday > 7 * nth) {
+        lastSunday -= 7;
+    }
+    return lastSunday;
+}
+
+static bool isDST(rtcTime_t t) {
+    dateTime_t dateTime;
+    rtcTimeToDateTime(&dateTime, t);
+    int lastSunday;
+    switch (timeConfig()->tz_dst_country) {
```

Comment:
Once you make `tz_automatic_dst` an uint8_t, cast it here to `tz_automatic_dst_e`, so the compiler will make sure the checks are exhaustive. i.e.

`switch ((tz_automatic_dst_e)timeConfig()-> tz_automatic_dst) {`